src/cpp/session/modules/SessionFiles.cpp

```diff
@@ -429,25 +429,30 @@ core::Error renameFile(const core::json::JsonRpcRequest& request,
    std::string path, targetPath;
    Error error = json::readParams(request.params, &path, &targetPath);
    if (error)
-      return error ;
+      return error;
 
+   // detect case-only name changes
+   bool isCaseOnlyChange =
```

Comment:
Do you think we should return an error (as we did before) when your "rename" doesn't change the name at all? I think we should because it's usually unintentional (although maybe better to catch it at the UI level).

Reply:
This will indeed still return an error -- the `isCaseOnlyChange` checks to see that the paths are only identical after lower-casing (and not identical before that)

Reply:
Thanks, I missed the first half of that conditional!